packages/satori-js/client.ts

```diff
@@ -125,14 +125,104 @@ export class Client {
     return this.apiClient.satoriGetExperiments(session.token, names);
   }
 
+  /** Get a single flag for this identity. */
+  async getFlag(session: Session, name: string, defaultValue?: string) {
+    try
+    {
+      if (this.autoRefreshSession && session.refresh_token &&
+        session.isexpired((Date.now() + this.expiredTimespanMs)/1000)) {
+        await this.sessionRefresh(session);
+      }
+
+      return this.apiClient.satoriGetFlags(session.token, "", "", [name]).then((flagList) => {
+        let flag = null;
+
+        flagList.flags?.forEach((f) => {
+          if (f.name === name) {
+            flag = f
+          }
+        });
+
+        if (flag === null) {
+          flag = {
+            name,
+            value: defaultValue
+          };
+        }
+
+        return Promise.resolve(flag);
+      });
+    }
+    catch (error) {
+      if (defaultValue !== undefined) {
+        return Promise.resolve({
+          name,
+          value: defaultValue
+        });
+      } else {
+        return Promise.reject(error);
+      }
+    }
+  }
+
+  /** Get a single flag with its configured default value. */
+  async getFlagDefault(session: Session, name: string, defaultValue?: string) {
```

Comment:
So there's some confusing terminology in Satori that we have to clear up:

`GetFlag(s)Default` actually refers to getting flags from the server before authenticating. This is needed in case anything like the authentication process itself or something before that process needs a/b testing.

Then there's passing a "default value" to a `getFlag` call, which is for returning a value on the client if it doesn't exist in the server.

Reply:
Removed the session param from this function.

Reply:
hmm I still see it here.